functorch/experimental/cond.py

```diff
@@ -151,7 +152,21 @@ def cond_python_dispatcher(*args):
     return cond(*args)
 
 
+@cond.py_impl(DispatchKey.FuncTorchDynamicLayerFrontMode)
+def cond_functionalize(pred, true_fn, false_fn, inputs):
+    functional_true_fn = functionalize(true_fn)
+    functional_false_fn = functionalize(false_fn)
+
+    _ = ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.FuncTorchDynamicLayerFrontMode))
+
+    if pred:
+        return functional_true_fn(*inputs)
+    return functional_false_fn(*inputs)
+
+
 # TODO(voz): Make this automatic for keys, this is very ugly atm
+cond.fallthrough(DispatchKey.Functionalize)
+cond.fallthrough(DispatchKey.FuncTorchDynamicLayerBackMode)
```

Comment:
We can stick with this approach to test things (my above suggestions) out, but when you want to ship this it would be good to override functorch.functionalize via TransformType.functionalize and actually write up the FunctionalizeInterpreter. FunctionalizeInterpreter should be roughly similar to what VmapInterpreter does.

Reply:
process function needs to promote non-functionaltensorwrapper into functionaltensorwrapper